vpc.tf

```
@@ -1,106 +1,101 @@
 resource "aws_vpc" "main" {
-  cidr_block       = "10.0.0.0/16"
+  cidr_block       = var.vpc_cidr
   instance_tenancy = "default"
 
   tags = {
-    Name = "main"
+    Name = "${var.env_code}-vpc"
   }
 }
 
-locals {
-  public_cidr = ["10.0.0.0/24", "10.0.1.0/24"]
-  private_cidr = ["10.0.2.0/24", "10.0.3.0/24"]
-}
-
-resource "aws_subnet" "public_subnet" {
-  count = length(local.public_cidr)
+resource "aws_subnet" "public" {
+  count = length(var.public_cidr)
 
-  vpc_id            = aws_vpc.main.id
-  cidr_block        = local.public_cidr[count.index]
+  vpc_id     = aws_vpc.main.id
+  cidr_block = var.public_cidr[count.index]
 
   tags = {
-    Name = "public-subnet ${count.index}"
+    Name = "${var.env_code}-public ${count.index}"
```

No need to have a whitespace betwen public and ${count.index}
```suggestion
    Name = "${var.env_code}-public${count.index}"
```
Also, please fix the same in other parts of the code